resources/frontend.rb

```diff
@@ -19,8 +19,18 @@
       variables['frontend'][new_resource.name] ||= {}
       variables['frontend'][new_resource.name]['default_backend'] ||= ''
       variables['frontend'][new_resource.name]['default_backend'] << new_resource.default_backend
-      variables['frontend'][new_resource.name]['bind'] ||= ''
-      variables['frontend'][new_resource.name]['bind'] << new_resource.bind
+      variables['frontend'][new_resource.name]['bind'] = []
+      if new_resource.bind.is_a? Hash
+        new_resource.bind.map do |addresses, ports|
+          addresses = Array(addresses) unless addresses.is_a? Array
+          ports = Array(ports) unless ports.is_a? Array
```

Comment:
```
addresses = Array(addresses) unless addresses.is_a? Array
ports = Array(ports) unless ports.is_a? Array
```

Not sure why my previous comment for this went away but those two lines are not needed.  You are already wrapping them vars on [L27](https://github.com/sous-chefs/haproxy/pull/205/files#diff-c5317915b0ca58ddec0a8c4e655c9d72R27)  and `Array()` will do the right thing if the passed in var is already an Array